pkg/components/prometheus-operator/component_conversion_test.go

```diff
@@ -0,0 +1,137 @@
+package prometheus //nolint:testpackage
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/hashicorp/hcl/v2"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	yamlserializer "k8s.io/apimachinery/pkg/runtime/serializer/yaml"
+	"k8s.io/client-go/util/jsonpath"
+
+	"github.com/kinvolk/lokomotive/pkg/components/util"
+)
+
+func renderManifests(t *testing.T, configHCL string) map[string]string {
+	name := "prometheus-operator"
+
+	component := newComponent()
+
+	body, diagnostics := util.GetComponentBody(configHCL, name)
+	if diagnostics.HasErrors() {
+		t.Fatalf("Getting component body: %v", diagnostics.Errs())
+	}
+
+	diagnostics = component.LoadConfig(body, &hcl.EvalContext{})
+	if diagnostics.HasErrors() {
+		t.Fatalf("Valid config should not return an error, got: %v", diagnostics)
+	}
+
+	ret, err := component.RenderManifests()
+	if err != nil {
+		t.Fatalf("Rendering manifests with valid config should succeed, got: %s", err)
+	}
+
+	return ret
+}
+
+//nolint:funlen
+func TestConversion(t *testing.T) {
```

Comment:
Maybe:
```suggestion
func TestManifestFields(t *testing.T) {
```

Reply:
Or `TestRenderManifests()` because the function we are testing is `RenderManifests()`.

Reply:
We already have a `TestRenderManifests` https://github.com/kinvolk/lokomotive/blob/60a34ccb935b6f1c7945a35c3184f701d21ed078/pkg/components/prometheus-operator/component_test.go#L24

Reply:
Then do we need a separate function? As far as I can tell the code in `TestConversion` tests that manifests are rendered properly...

Reply:
Yeah we do need, since it is doing more than just verify if a certain config is converted. It is matching specific input param to specific output param.